### ISSUE TITLE
Segmentation fault in X509_ALGOR_get0 during PKCS#12 processing 

### ISSUE THREAD

**Issue by MarcSchoenefeld** [openssl/openssl#26106]
When opening a PKCS#12 file with the pkcs12 command line tool, a null pointer deref happens in X509_ALGOR_set0. This was seen with version 3.4 and 3.5.0-dev. Version 3.0 does not seem to be affected and instead complains about a missing MAC for the PoC file. Filing as public bug, as requested after security triage, and https://github.com/openssl/openssl/pull/25144 was identified as regression cause. 

How to reproduce: 
 ````
(gdb) r
Starting program: /home/user/gits/openssl/apps/openssl pkcs12 -in X509_ALGOR_get0_testpassword.pk12 -passin pass:testpassword
Program received signal SIGSEGV, Segmentation fault.
0x00000000005327f9 in X509_ALGOR_get0 ()
(gdb) bt
#0  0x00000000005327f9 in X509_ALGOR_get0 ()
#1  0x00000000004314c0 in pkcs12_main ()
#2  0x000000000042cab2 in do_cmd ()
#3  0x000000000040240d in main ()
 
(gdb) disass $pc-10,$pc+10
Dump of assembler code from 0x5327ef to 0x532803:
   0x00000000005327ef:  nop
   0x00000000005327f0 <X509_ALGOR_get0+0>:      endbr64
   0x00000000005327f4 <X509_ALGOR_get0+4>:      test   rdi,rdi
   0x00000000005327f7 <X509_ALGOR_get0+7>:      je     0x5327ff <X509_ALGOR_get0+15>
=> 0x00000000005327f9 <X509_ALGOR_get0+9>:      mov    rax,QWORD PTR [rcx]
   0x00000000005327fc <X509_ALGOR_get0+12>:     mov    QWORD PTR [rdi],rax
   0x00000000005327ff <X509_ALGOR_get0+15>:     test   rsi,rsi
   0x0000000000532802 <X509_ALGOR_get0+18>:     je     0x53281d <X509_ALGOR_get0+45>
 
(gdb) info registers rcx rax
rcx            0x0                 0
rax            0x0                 0
 ````
[X509_ALGOR_get0_testpassword.pk12.gz](https://github.com/user-attachments/files/17997244/X509_ALGOR_get0_testpassword.pk12.gz)

**Comment by beldmit** [comment]
Will do but not earlier than on Friday

**Comment by KnowledgeReal69** [comment]
> [!IMPORTANT]
> Fix errors

**Comment by beldmit** [comment]
Should be fixed now, see #26140
